src/Microsoft.ServiceFabric.Actors.KVSToRCMigration/KVStoRCMigrationActorStateProvider.cs

```diff
@@ -382,6 +386,167 @@ public async Task<long> SaveStateAsync(List<KeyValuePair> kvsData, CancellationT
             return keysMigrated.Count;
         }
 
+        /// <summary>
+        /// Gets Value for given Key from RC state provider
+        /// </summary>
+        /// <param name="key">Key to fetch</param>
+        /// <param name="cancellationToken">cancellationToken</param>
+        /// <returns>Value for given key</returns>
+        public async Task<byte[]> GetValueByKeyAsync(string key, CancellationToken cancellationToken)
+        {
+            using (var tx = this.rcStateProvider.GetStateManager().CreateTransaction())
+            {
+                IReliableDictionary2<string, byte[]> dictionary = null;
+                if (key.StartsWith(ActorPresenceKeyPrefix))
+                {
+                    dictionary = this.rcStateProvider.GetActorPresenceDictionary();
+                }
+
+                // It is not right to assume the ActorId wouldn't have underscores.
+                // TODO: Handle this in ambiguous ActorId PR
+                else if (key.StartsWith(ActorStorageKeyPrefix))
+                {
+                    var startIndex = this.GetNthIndex(key, '_', 2);
+                    var endIndex = this.GetNthIndex(key, '_', 3);
+                    var actorId = new ActorId(key.Substring(startIndex + 1, endIndex - startIndex - 1));
+                    dictionary = this.rcStateProvider.GetActorStateDictionary(actorId);
+                }
+                else if (key.StartsWith(ReminderCompletedeStorageKeyPrefix))
+                {
+                    dictionary = this.rcStateProvider.GetReminderCompletedDictionary();
+                }
+                else if (key.StartsWith(ReminderStorageKeyPrefix))
+                {
+                    var startIndex = this.GetNthIndex(key, '_', 2);
+                    var endIndex = this.GetNthIndex(key, '_', 3);
+                    var actorId = new ActorId(key.Substring(startIndex + 1, endIndex - startIndex - 1));
+                    dictionary = this.rcStateProvider.GetReminderDictionary(actorId);
+                }
+                else
+                {
+                    var message = "Migration Validation Error: Failed to parse the KVS key - " + key;
```

Comment:
When would this happen? Does running into this block means data corruption, then we need to declare partition error.

Reply:
Yes, this means data corruption